src/vivarium_testing_utils/automated_validation/data_loader.py

```diff
@@ -108,16 +108,56 @@ def get_data(self, data_key: str, source: DataSource) -> Any:
     def upload_custom_data(self, data_key: str, data: pd.DataFrame) -> None:
         self._add_to_cache(data_key, DataSource.CUSTOM, data)
 
+    def cache_gbd_data(
+        self, data_key: str, data: pd.DataFrame, overwrite: bool = False
+    ) -> None:
+        """Upload or update a custom DataFrame or Series to the GBD context given by a data key."""
+        if data_key in self._raw_data_cache[DataSource.GBD] and not overwrite:
+            existing = self._raw_data_cache[DataSource.GBD][data_key]
```

Comment:
I would support putting stronger restrictions on this. Currently I believe the only data key that would be a string is location and it seems like we probably do not want users to be able to overwrite that. Furthermore, in the future, location will likely be a list of all the locations in a results directory.